### ISSUE TITLE
[Faceted Filters] Facet control missing attributes for context

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#107]
## Issue summary

Throughout the JS enabled experience, facet controls are marked up as `details`/`summary`, show/hide components. In some AT (VO, for example) in some contexts, this pairing could convey incorrect semantic meaning; hearing "summary" may not be expected.

In addition, some environments do not support `details`/`summary` well (as experience in a Fable usability test session when the control did not announce the current state consistently.)

The same issue applies for the mobile facet controls.

<details>
  <summary>Screenshots</summary>
<img width="948" alt="Screen shot of mobile faceted filters section. Facet controls are highlighted." src="https://user-images.githubusercontent.com/1392632/124605728-37c41980-de3a-11eb-9fbf-a1441aa2f1cc.png">

</details>

#### HTML

```html
<!-- Desktop… -->
<details
  class="disclosure-has-popup facets__disclosure js-filter"
  data-index="1"
>
  <summary class="facets__summary">
    <!-- … -->
  </summary>
  <div class="facets__display">
    <!-- … -->
  </div>
</details>

<!-- Mobile… -->
<details
  class="mobile-facets__details js-filter"
  data-index="mobile-1"
>
  <summary class="mobile-facets__summary">
    <!-- … -->
  </summary>
  <div class="mobile-facets__submenu">
    <!-- … -->
  </div>
</details>
```

## Steps to reproduce

1. Go to a collection page with Dawn theme with any supported browser.
2. With an active screen reader, navigate to a Filter control.

### Expected

- Controls to feature attributes conveying the [disclosure pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure).

### Actual

- Controls convey native `summary` semantics.

## Recommendation

**In the JS env only:**

1. Add the `role` attribute to the `summary` control, setting its value to `button`.
2. Add the `aria-expanded` attribute to the `summary` control, toggling its value from `false` to `true` upon activation.
3. Add the `aria-controls` attribute to the `summary` control, setting its value to the `id` of the `div` content container.

With this in place, screen reader users will encounter a more familiar pattern to interact with. In addition, more environments will provide better support for the conveyed pattern via explicit attribute state.

#### HTML

```html
<!-- Desktop… -->
<details
  class="disclosure-has-popup facets__disclosure js-filter"
  data-index="1"
>
  <summary class="facets__summary" role="button" aria-expanded="false" aria-controls="FacetContent0">
    <!-- … -->
  </summary>
  <div class="facets__display" id="FacetContent0">
    <!-- … -->
  </div>
</details>

<!-- Mobile… -->
<details
  class="mobile-facets__details js-filter"
  data-index="mobile-1"
>
  <summary class="mobile-facets__summary" role="button" aria-expanded="false" aria-controls="MobileFacetContent0">
    <!-- … -->
  </summary>
  <div class="mobile-facets__submenu" id="MobileFacetContent0">
    <!-- … -->
  </div>
</details>
```

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Faceted Filters
- **WCAG Principal:** Perceivable, Robust
- **WCAG SC:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html), [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
- **Severity:** High
- **Effort:** Medium
<!-- - **Related:** -->

**Comment by svinkle** [comment]
Fixed via #628.
